File: gs/src/main/java/com/pluxity/device/dto/GsDeviceResponse.java
package com.pluxity.device.dto;

import com.pluxity.feature.dto.FeatureResponse;
import lombok.Builder;

@Builder
public record GsDeviceResponse(
        String id,
        String name,
        FeatureResponse feature,
        DeviceCategoryResponseWithoutChildren deviceCategory) {}

File: gs/src/main/kotlin/com/pluxity/device/dto/GsDeviceResponse.kt
package com.pluxity.device.dto

import com.pluxity.device.GsDevice
import com.pluxity.feature.dto.FeatureResponse
import com.pluxity.file.dto.FileResponse

data class GsDeviceResponse(
    val id: String,
    val name: String,
    val feature: FeatureResponse?,
    val deviceCategory: DeviceCategoryResponseWithoutChildren?,
)

fun GsDevice.toGsDeviceResponse(thumbnailFile: FileResponse?): GsDeviceResponse =
    GsDeviceResponse(
        id = this.id,
        name = this.name,
        feature = this.feature?.let(FeatureResponse::from),
        deviceCategory = this.category?.let { DeviceCategoryResponseWithoutChildren.from(it, thumbnailFile) },
    )
